lib/improver/tests/lapse_rate/Test_LapseRate.py

```diff
@@ -0,0 +1,367 @@
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Unit tests for the LapseRate plugin."""
+
+import numpy as np
+import unittest
+
+import cf_units
+from iris.cube import Cube
+from iris.tests import IrisTest
+from iris.coords import (DimCoord,
+                         AuxCoord)
+
+from improver.grids import STANDARD_GRID_CCRS
+
+from improver.lapse_rate import LapseRate
+
+
+def reset_cube_data(temperature_cube, orography_cube, land_sea_mask_cube):
+    """ Resets the cube data to its defaults. """
+    temperature_cube.data[:, :, :] = 0
+    orography_cube.data[:] = 0
+    land_sea_mask_cube.data[:] = 1
+
+
+class Test__repr__(IrisTest):
+    """Test the repr method."""
+
+    def test_basic(self):
+        """Test that the __repr__ returns the expected string."""
+        result = str(LapseRate())
+        msg = ('<LapseRate>')
+        self.assertEqual(result, msg)
+
+
+class Test_calc_lapse_rate(IrisTest):
+    """Test the _calc_lapse_rate function."""
+
+    def setUp(self):
+        """Sets up arrays."""
+
+        self.temperature = np.array([280.06, 279.97, 279.90, 280.15, 280.03,
+                                     279.96, 280.25, 280.33, 280.27])
+        self.orography = np.array([174.67, 179.87, 188.46, 155.84, 169.58,
+                                   185.05, 134.90, 144.00, 157.89])
+
+    def test_returns_expected_values(self):
+        """Test that the function returns expected lapse rate. """
+
+        expected_out = -0.00765005774676
+        result = LapseRate()._calc_lapse_rate(self.temperature, self.orography)
+        self.assertArrayAlmostEqual(result, expected_out)
+
+    def test_handles_nan(self):
+        """Test that the function returns a blank value when central point
+           is NaN."""
+
+        self.temperature[4] = np.nan
+        expected_out = 0.0
+        result = LapseRate()._calc_lapse_rate(self.temperature, self.orography)
+        self.assertArrayAlmostEqual(result, expected_out)
+
```

Comment:
Can we have another test please to show what happens when some of the neighbourhood points are NaNs (ie where some do not meet the minimum height difference criterion)?